src/pinghelper.cpp

```diff
@@ -192,3 +201,25 @@ double PingHelper::loss() const {
   }
   return (double)(sendCount - recvCount) / sendCount;
 }
+
+void PingHelper::handlePingError() {
+  if (s_has_critical_ping_error) {
+    return;
+  }
+  logger.info() << "Encountered Unrecoverable ping error, switching to DUMMY "
+                   "Ping for next 10 Minutes";
+  // When the ping helper is unable to work, set the error flag
+  // and restart the pinghelper, to replace the impl with a dummy impl
+  // which we will use for the rest of the session
+  emit pingSentAndReceived(1);  // Fake a ping response with 1ms;
+  s_has_critical_ping_error = true;
+  stop();
+  start(m_gateway, m_source);
+
+  TimerSingleShot::create(this, 600000, [&] {  // 10 Minutes
+    logger.debug() << "Removing ping error state";
+    s_has_critical_ping_error = false;
+    stop();
+    start(m_gateway, m_source);
```

Comment:
What about if the time has been stopped in the meantime?

Reply:
good point lets just not do that :)